src/client/pfs.go

```diff
@@ -468,9 +469,10 @@ func (c APIClient) PutFileURL(repoName string, commitID string, path string, url
 		}
 	}()
 	if err := putFileClient.Send(&pfs.PutFileRequest{
-		File:     NewFile(repoName, commitID, path),
-		FileType: pfs.FileType_FILE_TYPE_REGULAR,
-		Url:      url,
+		File:      NewFile(repoName, commitID, path),
+		FileType:  pfs.FileType_FILE_TYPE_REGULAR,
+		Url:       url,
+		Recursive: recursive,
```

Comment:
It lets you do a recursively put the contents of an s3 directory. So if you have objects `dir/foo` and `dir/bar` in s3 then you can do:

```
pachctl put-file repo branch -r -f s3://dir
```

and it will put both `dir/foo` and `dir/bar`